aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -1057,8 +1069,6 @@ module aptos_framework::fungible_asset {
                 assert!(store.balance >= amount, error::invalid_argument(EINSUFFICIENT_BALANCE));
                 store.balance = store.balance - amount;
             };
-
```

Comment:
weren't folks complaining that there wasn't a Withdraw before? after migration now it is consistently to be always there?

also changing this after migration has started seems even more confusing? maybe we should create WithdrawForGas event and emit that one instead, so callers can choose what to do with it?

@movekevin for thoughts

Reply:
Part of the reason I break so many tests is due to this new event.
I don't know who complained there was not a withdraw before.
cc @kent-white @gregnazario for your thoughts as well.

Reply:
It seems mint_and_refund also has a deposit event... I am not sure whether we should have this PR now.

Reply:
can the caller know when (or how much) of the gas fee has been paid by coin (as so has no event) vs fa (and so has this event).
If that cannot be distinguished, then users cannot appropriately adjust the balances. so we should try to address that, in the cleanest way possible